sympy/solvers/solveset.py

```diff
@@ -847,6 +847,9 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if not free_symbols:
         b = Eq(f, 0)
+        if isinstance(symbol, Number):
```

Comment:
Okay then, I think in Line848 `if not free_symbols and getattr(symbol, 'is_Symbol', False)`  will work.